src/pages/AppList.svelte

```diff
@@ -1,28 +1,31 @@
 <script>
   import FilterInput from "../components/FilterInput.svelte";
+  import { fetchJSON } from "../state/api";
 
   const URL = "data/apps.json";
   let apps;
   let filteredApps;
-  fetch(URL)
-    .then((r) => r.json())
-    .then((ret) => {
-      apps = ret.sort((a, b) => a.app_id > b.app_id);
-      filteredApps = apps;
-    });
+
+  const filteredDataPromise = async () => {
+    apps = await fetchJSON(URL);
+    apps = apps.sort((a, b) => a.app_id > b.app_id);
+    return apps;
+  };
+
+  filteredApps = filteredDataPromise();
 
   function filterApps(filterText) {
     filteredApps = apps.filter((appItem) => appItem.name.includes(filterText));
   }
 </script>
 
 <h2>Applications</h2>
-<FilterInput onChangeText={filterApps} />
-{#if apps}
-  {#each filteredApps as app}
+{#await filteredApps then apps}
```

Comment:
The variable assignment flow here is confusing, especially since `filteredApps` can be both an async function (its initial assignment) and an array (assignment from `filterApps`). `filteredApps` also assigns at the same time as it returns a variable.

I'd actually recommend using the `onMount` function here, which should reduce the size of the diff.

Reply:
Yes, the variable flow is confusing, and naming things is very difficult. I will look into onMount and add the suggested changes. Thanks.